sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -107,6 +109,31 @@ public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
             .doOnError(throwable -> logger.warning("<-- HTTP FAILED: ", throwable));
     }
 
+    @Override
+    public HttpResponse processSynchronously(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // No logging will be performed, trigger a no-op.
+        if (httpLogDetailLevel == HttpLogDetailLevel.NONE) {
+            return next.processSynchronously();
+        }
+
+        final ClientLogger logger = new ClientLogger((String) context.getData("caller-method").orElse(""));
+        final long startNs = System.nanoTime();
+
+        requestLogger.logRequestSynchronously(logger, getRequestLoggingOptions(context));
+        try {
+            HttpResponse response = next.processSynchronously();
+            if (response != null) {
+                response = responseLogger.logResponseSynchronously(
+                    logger, getResponseLoggingOptions(response, startNs, context));
+            }
+            return response;
+        } catch (RuntimeException e) {
```

Comment:
This should be catching any exception fwiw, that is what `doOnError` does

Reply:
I would say reactor implementation is overreaching in this case (for good reasons perhaps because they'd convert any try catch into error emission). In this case however, the `next.processSynchronously()` doesn't declare any checked exceptions which leaves only two possibilities - either it's a RuntimeException or an Error.
I think we shouldn't be really logging Errors but I'm going to leave a todo to follow up on that later.